st2api/st2api/app.py

```diff
@@ -67,34 +49,22 @@ def setup_app(config=None):
                      register_signal_handlers=True,
                      register_internal_trigger_types=True,
                      run_migrations=True,
-                     config_args=config.config_args)
-
-    if not config:
-        # standalone HTTP server case
-        config = _get_pecan_config()
-    else:
-        # gunicorn case
-        if is_gunicorn:
-            config.app = _get_pecan_config().app
-
-    app_conf = dict(config.app)
+                     config_args=config.get('config_args', None))
 
-    active_hooks = [hooks.RequestIDHook(), hooks.JSONErrorResponseHook(),
-                    hooks.LoggingHook()]
+    router = Router(debug=cfg.CONF.api.debug, auth=cfg.CONF.auth.enable)
 
-    active_hooks.append(hooks.AuthHook())
-    active_hooks.append(hooks.CorsHook())
-
-    app = pecan.make_app(app_conf.pop('root'),
-                         logging=getattr(config, 'logging', {}),
-                         hooks=active_hooks,
-                         **app_conf
-                         )
+    spec = spec_loader.load_spec('st2common', 'openapi.yaml')
+    transforms = {
+        '^/api/v1/': ['/', '/v1/'],
+        '^/api/exp/': ['/exp/']
+    }
+    router.add_spec(spec, transforms=transforms)
 
-    # Static middleware which servers common static assets such as logos
-    static_root = os.path.join(BASE_DIR, 'public')
-    app = StaticFileMiddleware(app=app, directory=static_root)
+    app = router.as_wsgi
 
-    LOG.info('%s app created.' % __name__)
+    app = CorsMiddleware(app)
```

Comment:
Sure it does. If you look at how middleware work, they just call each other in sequence.